src/services/state-manager/state-manager.service.ts

```diff
@@ -0,0 +1,29 @@
+import _ from "lodash"
+
+export class StateManagerService {
+  private static _instance: StateManagerService
+  private _currentBotState: string = 'Normal'
+  private _currentBotStep: number = 0
```

Comment:
If you know before hand the range of the step (meaning you that there is for example 5 steps) you could use an enum instead

Reply:
Nope, I don't know the range of steps. It depends on the command/state and I could still add some.

Reply:
Yes but how many step do you think there is?
I am trying to help you decide if an enum you be a good place or not.
If the quantity of data is small enough and predefined an enum should be the type to use.

Reply:
So, the step depends on the command and subcommand, really. It's used when people type in value directly and it really depends on how much data they have to type in.

Thus this answers your question about data being an enum or not xD

Reply:
Well then it seems that you should keep number type then.

Reply:
@asmodee-bellanie check this comment I am merging